src/cephadm/cephadm

```diff
@@ -108,13 +108,25 @@ def port_in_use(port_num):
     # type (int) -> bool
     """Detect whether a port is in use on the local machine - IPv4 and IPv6"""
 
+    def attempt_bind(address):
+        # type (str) -> None
+        try:
+            s.bind((address, port_num))
+        except (socket.error, OSError) as e:  # py2 and py3
+            if e.errno == 98:                 # Address already in use
+                raise OSError
+            elif e.errno == 99:               # Cannot assign requested address
+                return
+
     try:
         s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        s.bind(("127.0.0.1", port_num))
+        attempt_bind('0.0.0.0')
         s.close()
+
         s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
-        s.bind(("::1", port_num))
+        attempt_bind('::/0')
         s.close()
+
     except OSError:
         s.close()
```

Comment:
```suggestion
    finally:
        s.close()
```

Reply:
Close the socket in one place via `finally` block ?

Reply:
Done. close moved to inner function's finally block